CMakeLists.txt

```diff
@@ -85,8 +85,23 @@ include(ROCMPackageConfigHelpers)
 include(ROCMInstallSymlinks)
 include(CheckCXXCompilerFlag)
 
+# check if asan is enabled
+if (NOT DEFINED ADDRESS_SANITIZER AND DEFINED ENV{ADDRESS_SANITIZER})
+    set(ADDRESS_SANITIZER $ENV{ADDRESS_SANITIZER})
+endif()
+if (ADDRESS_SANITIZER OR CMAKE_CXX_FLAGS MATCHES "-fsanitize=address")
+    set(ADDRESS_SANITIZER_ENABLED ON)
+else()
+    set(ADDRESS_SANITIZER_ENABLED OFF)
+endif()
+
+if (ADDRESS_SANITIZER_ENABLED)
+rocm_check_target_ids(DEFAULT_AMDGPU_TARGETS
+  TARGETS "gfx908:xnack+;gfx90a:xnack+;gfx940:xnack+;gfx941:xnack+;gfx942:xnack+" )
+else()
 rocm_check_target_ids(DEFAULT_AMDGPU_TARGETS
   TARGETS "gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+;gfx940;gfx941;gfx942" )
```

Comment:
For the regular targets, we can remove all xnack specification. E.g., gfx908;gfx90a;gfx940;...
This gives us 1 less target